Assets/Scripts/Controllers/HistoryController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Controllers;
+using Services.History;
+
+namespace Controllers
+{
+    public class HistoryController: MonoBehaviour
+    {
+        private Stack<ICommand> _Undocommands = new Stack<ICommand>();
+        private Stack<ICommand> _Redocommands = new Stack<ICommand>();
+        private NodeController nodeController;
+
+        void Awake()
+        {
+            nodeController = GetComponent<NodeController>();
+        }
+
+        public void Redo(int levels)
+        {
+            for (int i = 1; i <= levels; i++)
+            {
+                if (_Redocommands.Count != 0)
+                {
+                    ICommand command = _Redocommands.Pop();
+                    command.Execute();
+                    _Undocommands.Push(command);
+                }
+
+            }
+        }
+
+        public void Undo(int levels)
+        {
+            for (int i = 1; i <= levels; i++)
+            {
+                if (_Undocommands.Count != 0)
+                {
+                    ICommand command = _Undocommands.Pop();
+                    command.UnExecute();
+                    _Redocommands.Push(command);
+                }
+
+            }
+        }
+
+        #region UndoHelperFunctions
+
+        public void InsertInUnDoRedoForJump(Model.Node prevNode, Model.Node targetNode)
```

Comment:
Generify:
- Use more meaningful name like `RegisterAction`
- Pass a created action `ICommand` argument (this function should be used to register any action)
```suggestion
        public void RegisterAction(ICommand action)
```